### PR TITLE
WA-268 Account locking/unlocking

### PR DESCRIPTION
**Pull request by germartinez**
The user is allowed to lock/unlock the Ethereum account inside the Chrome extension.
- When the extension is locked it is required to write the password every time the user wants to do a certain actions like adding a new safe or sign a transaction.
- When the extension is unlocked, the user don't need to write the password for the actions described above

### CODE REVIEW

File: app/components/Page/index.js [hunk 1]
```diff
@@ -1,23 +1,35 @@
...
-    }
-    {props.whitelist && <WhitelistedDappState />}
-    <div className={styles.container}>
-      {props.children}
+const cx = classNames.bind(styles)
+
+const Page = ({
+  children,
+  padding,
+  ...props,
```
Comment (apanizo) [inline app/components/Page/index.js]: That is dangerous. You are passing here props without a name. As far as I see they are used on the Header component.

I suggest transforming the Header component into something like [this](https://github.com/gnosis/gnosis-team-safe/tree/master/src/components/Header)

In that way to do not overload the Page component with props only related to Header component, and it is the Header the responsible of fetching stuff from the redux store using its own selectors.

Also, your project is getting bigger, I strongly recommend to introduce a type checker (Flow is my choice).

File: app/routes/Account/components/LockingState/index.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    this.props.onUnlockAccount(mnemonic, unlockingTime.toISOString())
+  }
+
+  render() {
+    const { account } = this.props
+
+    return (
+      <div className={styles.lockedState}>
+        {(account.lockedState)
+          ?
```
Comment (apanizo) [inline app/routes/Account/components/LockingState/index.js]: Difficult to read. I recommend to extract in this class doing something like:
```
const MyComponent = ({...}) => ( ...)
const MyOtherComponent = ({...}) => ( ...)

class LockingState extends Component {
     ....
    render () {
         ...
          account.lockedState ? <MyComponent /> : <MyOtherComponent />   <--- With meaningful names
    }
} 
```

File: app/routes/Account/components/LockingState/index.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+          <div className={styles.lockedAccounts}>
+            <div className={styles.state}>Safes are locked</div>
+            <Link to={{
+              pathname: '/password',
+              state: {
```
Comment (apanizo) [inline app/routes/Account/components/LockingState/index.js]: What it does ```state: dest: '/account'```?

File: app/routes/SafesList/components/Layout.js [hunk 4]
```diff
@@ -9,18 +9,26 @@ class Layout extends Component {
...
-          state: {
-            dest: '/pairing'
-          }
-        }}>
-          <button>Connect to safe</button>
-        </Link>
+      <Page account logOut padding='noPadding'>
+        {lockedState
+          ?
+          <Link to={{
```
Comment (apanizo) [inline app/routes/SafesList/components/Layout.js]: If the only difference is the ```to``` object I suggest to only evaluate it separately. Something like:
```
<Page>
    <Link to={ lockedState ? passwordRoute : pairingRoute }>
        <button> .... </button>
    </Link>
</Page>
```

File: app/routes/Settings/components/Layout.js [hunk 5]
```diff
@@ -7,7 +7,7 @@ import styles from './index.css'
 class Layout extends Component {
   render() {
     return (
-      <Page account logOut>
+      <Page account logOut padding='noPadding'>
```
Comment (apanizo) [inline app/routes/Settings/components/Layout.js]: if the most common situation now is having the <Page> component without padding consider change the component removing the padding making it optional

File: app/routes/PairingProcess/utils/ethOperations.js [hunk 6]
```diff
@@ -13,12 +13,13 @@ export const createAccountFromMnemonic = (mnemonic) => {
 
 export const generatePairingCodeContent = (privateKey) => {
   const startDate = new Date()
-  const expiryDate = new Date(startDate.setMinutes(startDate.getMinutes() + 10))
+  const expirationDate = new Date(startDate.setMinutes(startDate.getMinutes() + 10))
```
Comment (apanizo) [inline app/routes/PairingProcess/utils/ethOperations.js]: I see this strange. Maybe something like:
```const expirationDate = new Date().setMinutes(startDate.getMinutes() + 10)```

Comment (germartinez) [reply app/routes/PairingProcess/utils/ethOperations.js]: Your proposal is not the same as what is written in the code. The former returns milliseconds while the later returns a Date object.

File: app/routes/Settings/components/Layout.js (hunk 5 above)
Comment (germartinez) [reply app/routes/Settings/components/Layout.js]: The most common situation is having the Page component with padding

File: app/routes/Account/components/LockingState/index.js (hunk 3 above)
Comment (germartinez) [reply app/routes/Account/components/LockingState/index.js]: "dest" param is used to tell the Password component where to go after finishing. This way the Password component can redirect to a different page depending on the screen flow where it is used.

**Pull request merged**